client/src/pages/private/CohortDetails.js

```diff
@@ -69,7 +69,16 @@ export default ({ match }) => {
     }
   };
 
-  const getParticipantGraduationStatus = (participantStatuses) => {
+  const getParticipantGraduationStatus = (participantStatuses, participantId, participants) => {
+    // check if any of the participants were re-assigned to the same cohort
+    if (
+      participants.filter((participant) => participant.participant_id === participantId).length > 1
+    ) {
+      // extra validation for the edge case
+      if (!participantStatuses.find((postHireStatus) => postHireStatus.is_current === true)) {
+        return getPostHireStatusLabel(participantStatuses[participantStatuses.length - 1]);
+      }
```

Comment:
I'm finding the logic here a bit difficult to follow.
For the conditions, would prefer to see something like:
```
const numDuplicateParticipants = participants.filter...
if (numDuplicateParticipants > 0) {....
const currentPostHireStatuses = ...
if (!currentPostHireStatuses)...
```
where we have a meaningful variable names assigned so it's easier to read the business logic from the code.